openedx/openedx-platform
# XBlock Reporting: Display a popup after requesting a report

This feature implements a popup for generating / downloading the xblock report. It will poll for the generation task status and after it's completed, it will show a button to download the report directly in the modal window.

**Dependencies**: https://github.com/edx/edx-platform/pull/18170

**Screenshots**: 
![screenshot-2018-5-21 instructor dashboard your platform name here](https://user-images.githubusercontent.com/1730454/40352449-244c4062-5daf-11e8-8463-aa4ddad9ab28.png)

**Sandbox URL**: TBD - sandbox is being provisioned.

**Merge deadline**: None

**Testing instructions**:

1. Make sure `ENABLE_GRADE_DOWNLOADS` is enabled in lms/djangoapps/instructor/settings/common.py
2. Go to Instructor Dashboard -> Data Downloads
3. In section "Select a problem to generate a CSV file that lists all student answers to the problem. You also select a section or chapter to include results of all problems in that section or chapter."
  - Pick a course block
  - Click "Create a report of problem responses"
  - Observe the modal window

**Reviewers**
